podaac/subsetter/xarray_enhancements.py

```diff
@@ -80,14 +80,14 @@ def get_indexers_from_nd(cond, cut):
     if not np.any(rows) | np.any(cols):
         logging.info("No data within the given bounding box.")
 
+    cond_list = [dim for dim in cond.dims if "time" not in dim ]
```

Comment:
Is there any way we can pass in the name of the time dimension instead of having to hardcode the string here? I think we have more sophisticated "dimension finding" logic elsewhere in the module that we might be able to use?

Reply:
Can we use this function instead https://github.com/podaac/l2ss-py/blob/develop/podaac/subsetter/subset.py#L456  ?

Reply:
Ok, the time variables should be able to imported as well

Reply:
@frankinspace worth the discussion. But I think 'time' does need to be hard coded in because the get_time_var doesn't necessarily get a time dimension. For tropomi's case it gets delta_time but time is the dimension that goes through to get_indexers_nd. There is one case that has ddm as a spatial dimension and ddm_timeutc as the time dimension. A if dim != time_var_name won't work because of the tropomi delta_time, and a dim is in time_var_name also won't work because of this ddm and ddm_timeutc. Only containing 'time' works for all cases currently

Reply:
hmm, yes I see. I'm just worried about the case where there is a dimension representing time that is called `t` or `i` or something other than word that includes `time`. We haven't actually run into that issue yet though so this is probably ok for now.

Is the fact that it is a time dimension the actual problem or is it that there is a third dimension of size 1? I wonder if treating it more like that would make this more robust?

Reply:
I think it a third dimension of size 1 which is why another squeeze() method is applied on the rows and cols. Maybe the cond.dims just need to be squeezed. Will check that

Reply:
@frankinspace Updated to code to make sure a shape with length of one doesn't go through spatial dimensions

Reply:
Thanks @nlenssen2013 I think that makes a lot of sense and will hopefully be a bit more robust. Nice job!